main/helper.js

```diff
@@ -0,0 +1,245 @@
+  const result = await client.depositEther(amount, userAddress);
+  const txHash = await result.getTransactionHash();
+  console.log('Deposit ETH txHash = ', txHash);
+  const txReceipt = await result.getReceipt();
+
+  console.log('ETH deposited');
+};
+const burnETH = async (client, tokenAddressETH, amount) => {
+  const erc20Token = client.erc20(tokenAddressETH);
+  const result = await erc20Token.withdrawStart(amount);
+  const txHash = await result.getTransactionHash();
+  console.log('Burn ETH txHash = ', txHash);
+  const txReceipt = await result.getReceipt();
+
+  return txHash;
+};
+const withdrawETH = async (client, tokenAddressETH, txnHash) => {
+  const erc20RootToken = client.erc20(tokenAddressETH, true);
+  const result = await erc20RootToken.withdrawExit(txnHash);
+  const txHash = await result.getTransactionHash();
+  console.log('Withdraw ETH txHash = ', txHash);
+  const txReceipt = await result.getReceipt();
+};
+
+// --------------------------------------------------------------------------------------------------------------------------------------------- //
+
+// ETH -> Polygon for erc20
+
+const approveERC20 = async (erc20RootToken, amount) => {
+  const approveResult = await erc20RootToken.approve(amount);
+  const txHash = await approveResult.getTransactionHash();
+  console.log('Approve ERC20 txHash = ', txHash);
+  const txReceipt = await approveResult.getReceipt();
+
+  console.log('ERC20 approved');
+};
+const depositERC20 = async (erc20RootToken, amount, userAddress) => {
+  const result = await erc20RootToken.deposit(amount, userAddress);
+  const txHash = await result.getTransactionHash();
+  console.log('Deposit ERC20 txHash = ', txHash);
+  const txReceipt = await result.getReceipt();
+
+  console.log('ERC20 deposited');
+};
+const burnERC20 = async (client, tokenAddress, amount) => {
+  const erc20Token = client.erc20(tokenAddress);
+  const result = await erc20Token.withdrawStart(amount);
+  const txHash = await result.getTransactionHash();
+  console.log('Burn ERC20 txHash = ', txHash);
+  const txReceipt = await result.getReceipt();
+
+  return txHash;
+};
+const withdrawERC20 = async (client, tokenAddress, txHash) => {
+  const erc20RootToken = client.erc20(tokenAddress, true);
+  const result = await erc20RootToken.withdrawExit(txHash);
+  const txHashExit = await result.getTransactionHash();
+  console.log('Exit ERC20 txHash = ', txHashExit);
+  const txReceipt = await result.getReceipt();
+};
+
+// --------------------------------------------------------------------------------------------------------------------------------------------- //
+
+async function calculateBlockNum(timeInterval) {
+  const provider = new ethers.providers.EtherscanProvider('goerli');
+  const currentBlock = await provider.getBlockNumber();
+  const blockTime = 15; // ETH block time is 15 seconds
+
+  //Block number 5 mins ago
+  const block = currentBlock - timeInterval / blockTime;
+
+  return [block, currentBlock];
+}
```

Comment:
Instead of doing it strictly for 5mins, we can have it for 128 blocks.
block = currentBlock - 128

Reply:
https://github.com/Rushanksavant/Bridging/blob/2e5edd7a88b00d5d8600d661c7fff58f73827f54/main/depositTxn.js#L22

The 5 min is not fixed strictly, the caller can choose the time interval he wants.

The `execute()` function is exported as `main()` and used in cronJob with `recursiveInterval = 300 seconds`(5 min) :
https://github.com/Rushanksavant/Bridging/blob/2e5edd7a88b00d5d8600d661c7fff58f73827f54/cronJob.js#L11